### PR TITLE
feat(gatsby-plugin-image): Add support for backgroundColor in sharp

### PR DESCRIPTION
**Pull request by ascorbic**
Unifies API to support backgroundColor on both the component and the resolver. This is a breaking API change, as poreviously the resolver took `background`. However that was not actually implemented, so was ignored.

New behaviour is to pass the prop to sharp, so when resizing with a crop fit of e.g. `contain`, sharp uses that as the background color for letterboxing. The value is also set as the background color of the container, meaning that if the image has transparency, if there is no placeholder, or the placeholder is tracedSVG then the background is visible.

[ch22667]

### CODE REVIEW

File: packages/gatsby-plugin-sharp/src/image-data.ts [hunk 1]
```diff
@@ -118,11 +119,11 @@ export async function generateImageData({
     reporter.warn(
       `Specifying fullWidth images will ignore the width and height arguments, you may want a constrained image instead. Otherwise, use the breakpoints argument.`
     )
-    args.width = metadata.width
+    args.width = metadata?.width
```
Comment (laurieontech) [inline packages/gatsby-plugin-sharp/src/image-data.ts:122]: Is there a scenario where metadata will be undefined entirely? It looks like in the worst case it's an empty object.

File: packages/gatsby-plugin-sharp/src/image-data.ts [hunk 2]
```diff
@@ -283,14 +286,12 @@ export async function generateImageData({
...
-        quality,
-        ...transformOptions,
-        fit,
-        cropFocus,
+        ...sharedOptions,
         ...args.webpOptions,
         width,
         height: Math.round(width / imageSizes.aspectRatio),
         toFormat: `webp`,
+        background: backgroundColor,
```
Comment (laurieontech) [inline packages/gatsby-plugin-sharp/src/image-data.ts:294]: Should we passing this if we already have sharedOptions? That seems redundant.

File: packages/gatsby-plugin-sharp/src/image-data.ts [hunk 3]
```diff
@@ -317,13 +318,12 @@ export async function generateImageData({
...
       args: {
+        ...sharedOptions,
         ...options,
-        ...transformOptions,
-        fit,
-        cropFocus,
         toFormatBase64: args.blurredOptions?.toFormat,
         width: placeholderWidth,
         height: Math.round(placeholderWidth / imageSizes.aspectRatio),
+        background: backgroundColor,
```
Comment (laurieontech) [inline packages/gatsby-plugin-sharp/src/image-data.ts:326]: Same comment as above. sharedOptions is already passed.

File: packages/gatsby-transformer-sharp/src/customize-schema.js [hunk 4]
```diff
@@ -504,9 +504,8 @@ const imageNodeType = ({
         type: TransformOptionsType,
         description: `Options to pass to sharp to control cropping and other image manipulations.`,
       },
-      background: {
+      backgroundColor: {
```
Comment (laurieontech) [inline packages/gatsby-transformer-sharp/src/customize-schema.js:507]: Did we have any other doc where this needs to be changed? I don't think so, but wanted to mention.

**Review by laurieontech — COMMENTED** [review]
Overall looks solid. Left a few comments but should be good when those are addressed.

File: packages/gatsby-plugin-sharp/src/image-data.ts (hunk 2 above)
Comment (ascorbic) [reply packages/gatsby-plugin-sharp/src/image-data.ts:294]: Oops. Left that in after I refactored the others into sharedOptions

File: packages/gatsby-plugin-sharp/src/image-data.ts (hunk 1 above)
Comment (ascorbic) [reply packages/gatsby-plugin-sharp/src/image-data.ts:122]: If sharp throws an error getting metadata it would be undefined

Comment (ascorbic) [reply packages/gatsby-plugin-sharp/src/image-data.ts:122]: I'll change it so it's an empty object though

File: packages/gatsby-transformer-sharp/src/customize-schema.js (hunk 4 above)
Comment (ascorbic) [reply packages/gatsby-transformer-sharp/src/customize-schema.js:507]: No, I don't think we mention it

**Review by laurieontech — APPROVED** [review]
Nice work!

**Pull request merged**